lisp/ess-inf.el

```diff
@@ -1002,16 +1003,24 @@ Returns nil if TIMEOUT was reached, non-nil otherwise."
     (< elapsed timeout)))
 
 (defun inferior-ess-ordinary-filter (proc string)
-  (let ((delim (process-get proc 'output-delimiter)))
-    (unless delim
+  (ess--if-verbose-write-process-state proc string "ordinary-filter")
+  (let ((cmd-delim (process-get proc 'cmd-output-delimiter))
+        (cmd-async-restore-alist (process-get proc 'cmd-async-restore-alist))
+        (flush (lambda () (with-current-buffer (process-buffer proc)
+                       (insert string)))))
+    (if cmd-delim
+        (progn
+          (funcall flush)
+          (inferior-ess--set-status-sentinel proc (process-buffer proc) cmd-delim)
```

Comment:
I would still say that while things are changing so much now, let's not be redundant here. Basically, 

```
(inferior-ess--set-status proc string)
(inferior-ess-run-callback proc string)
(when cmd-delim 
  ... do cmd specific stuff)
```
Kinda want to figure this part while it's hot now, and not break user's code later.